Mathlib/NumberTheory/Padics/Complex.lean

```diff
@@ -0,0 +1,194 @@
+instance : Algebra (PadicAlg p) ℂ_[p] := UniformSpace.Completion.algebra' _
+
+theorem coe_eq (x : PadicAlg p) : (x : ℂ_[p]) = algebraMap (PadicAlg p) ℂ_[p] x := rfl
+
+theorem coe_zero : ((0 : PadicAlg p) : ℂ_[p]) = 0 := rfl
+
+/-- `ℂ_[p]` is an algebra over `ℚ_[p]`. -/
+instance : Algebra ℚ_[p] ℂ_[p] :=
+  ((algebraMap (PadicAlg p) ℂ_[p]).comp (algebraMap ℚ_[p] (PadicAlg p))).toAlgebra
+
+instance : IsScalarTower ℚ_[p] (PadicAlg p) ℂ_[p] := IsScalarTower.of_algebraMap_eq (congrFun rfl)
+
+@[simp, norm_cast]
+lemma coe_natCast (n : ℕ) : ((n : PadicAlg p) : ℂ_[p]) = (n : ℂ_[p])  := by
+  rw [← map_natCast (algebraMap (PadicAlg p) ℂ_[p]) n, coe_eq]
+
+/-- The valuation of `p : ℂ_[p]` is `1/p`. -/
+theorem valuation_p : Valued.v (p : ℂ_[p]) = 1 / (p : ℝ≥0) := by
+  have := valuation_extends p (p : PadicAlg p)
+  rw [PadicAlg.valuation_p] at this
+  simpa only [← map_natCast (algebraMap (PadicAlg p) ℂ_[p]), ← coe_eq]
+
+/-- The valuation on `ℂ_[p]` has rank one. -/
+instance : RankOne (PadicComplex.valued p).v where
+  hom         := MonoidWithZeroHom.id ℝ≥0
+  strictMono' := strictMono_id
+  nontrivial' := by
+    use p
+    haveI hp : Nat.Prime p := hp.1
+    simp only [valuation_p, one_div, ne_eq, inv_eq_zero, Nat.cast_eq_zero, inv_eq_one,
+      Nat.cast_eq_one]
+    exact ⟨hp.ne_zero, hp.ne_one⟩
+
+/-- `ℂ_[p]` is a normed field, where the norm corresponds to the extension of the `p`-adic
+  valuation. -/
+instance C_p.NormedField : NormedField ℂ_[p] :=  Valued.toNormedField _ _
+
+instance : NormedField (UniformSpace.Completion (PadicAlg p)) := C_p.NormedField p
+
+theorem norm_def : (Norm.norm : ℂ_[p] → ℝ) = Valued.norm := rfl
+
+/-- The norm on `ℂ_[p]` extends the norm on `PadicAlg p`. -/
+theorem norm_extends (x : PadicAlg p) : ‖(x : ℂ_[p])‖ = ‖x‖ := by
+  by_cases hx : x = 0
+  · rw [hx, coe_zero, norm_zero, norm_zero]
+  · rw [norm_def, Valued.norm, ← coe_nnnorm, ← PadicAlg.valuation_def, ← valuation_extends p x]
+    rfl
+
+/-- The `ℝ≥0`-valued norm on `ℂ_[p]` extends that on `PadicAlg p`. -/
+theorem nnnorm_extends (x : PadicAlg p) : ‖(x : ℂ_[p])‖₊ = ‖x‖₊ := by ext; exact norm_extends p x
+
+/-- The norm on `ℂ_[p]` is nonarchimedean. -/
+theorem isNonarchimedean : IsNonarchimedean (Norm.norm : ℂ_[p] → ℝ) := fun x y ↦ by
+  refine UniformSpace.Completion.induction_on₂ x y
+    (isClosed_le (Continuous.comp continuous_norm continuous_add)
+      (Continuous.max (Continuous.comp (@continuous_norm ℂ_[p] _) (Continuous.fst continuous_id))
+      (Continuous.comp (@continuous_norm ℂ_[p] _) (Continuous.snd continuous_id)))) (fun a b ↦ ?_)
+  · rw [← UniformSpace.Completion.coe_add, norm_extends, norm_extends, norm_extends]
+    exact PadicAlg.isNonarchimedean p a b
+
+end PadicComplex
+
+/-- We define `𝓞_ℂ_[p]` as the valuation subring of of `ℂ_[p]`, consisting of those elements with
+  valuation `≤ 1`. -/
+def padicComplexIntegers : ValuationSubring ℂ_[p] := (PadicComplex.valued p).v.valuationSubring
```

Comment:
All of a sudden the "p" is small instead of capitalized. What is the library consensus about this capitalization?

Reply:
Renamed to `PadicComplexInt`.